### PR TITLE
Cas 438 persist pdu id to applications table

### PR DESCRIPTION
**Pull request by danhumphreys-moj**
Ref: CAS-438, https://dsdmoj.atlassian.net/browse/CAS-438

As a part of the CAS-307: Capture PP's PDU and other core information in referral from Delius work, we are now planning to record and save the ID of the probation delivery unit (PDU).

In ticket CAS-366: FE: Implement new user journey in FE with pre-populated nDelius data the form will now return the PDU id as first class field called probationDeliveryUnitId. This needs to be saved into the db alongside the name of the PDU (determinable via the existing probation_delivery_units table).

Create a new probation_delivery_unit_id field and save the value returned from the UI probationDeliveryUnitId in it.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationTest.kt
Comment (danhumphreys-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationTest.kt:2683]: Presumably this is the right way to insert a PDU and region for integration testing?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/ApplicationEntity.kt
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/ApplicationEntity.kt:513]: I would expect this to be defined as a ManyToOne relationship to the ProbationDeliveryUnitEntity instead of a UUID. Is there a reason we haven't modelled it this way? As long as the ManyToOne annotation marks the relationship as fetch = lazy, there should be any performance penalty adding this (we've learnt that the hard way with spring boot 3 upgrade 😄 ) . The column name can remain as is.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationTest.kt
Comment (davidatkinsuk) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationTest.kt:2389]: Yep, i think that'll work fine 👍

**Review by davidatkinsuk — CHANGES_REQUESTED** [review]
Thanks Dan, I think there's one comment that needs considering before approval

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/ApplicationEntity.kt
Comment (danhumphreys-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/ApplicationEntity.kt]: Super, thanks for this.  Should be ok now?

Comment (davidatkinsuk) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/ApplicationEntity.kt]: Looks good!

**Review by davidatkinsuk — APPROVED** [review]
LGTM, thanks for making the change

**Pull request merged**
